src/views/CardStack.js

```diff
@@ -189,18 +182,27 @@ class CardStack extends Component<DefaultProps, Props, void> {
     transitionProps: NavigationTransitionProps,
     headerMode: HeaderMode
   ): ?React.Element<*> {
-    const headerConfig = this.props.router.getScreenConfig(
+    const headerConfig: HeaderConfig = this.props.router.getScreenConfig(
       transitionProps.navigation,
       'header'
     ) || {};
 
+    if (
+      headerConfig.bar === null
+      || (headerConfig.bar && typeof headerConfig.bar !== 'function')
+    ) {
+      return headerConfig.bar;
+    }
+
+    const HeaderComponent = headerConfig.bar || Header;
+
     return (
-      <this.props.headerComponent
+      <HeaderComponent
```

Comment:
Can you call this as a function please? As I said previously, this will result in unmounting and re-mounting the header everytime something changes when config is specified as following:

```
header: () => ({
  bar: props => <MyComponent {...props} />,
})
```